reference/configuration/framework.rst

```diff
@@ -169,39 +293,431 @@ see :doc:`/cookbook/request/load_balancer_reverse_proxy`.
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:trusted-host>.*\.?acme.com$</framework:trusted-host>
+                <framework:trusted-host>.*\.?acme.org$</framework:trusted-host>
+                <!-- ... -->
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/config.php
+        $container->loadFromExtension('framework', array(
+            'trusted_hosts' => array('.*\.?acme.com$', '.*\.?acme.org$'),
+        ));
+
+In addition, you can also set the trusted hosts in the front controller using
+the ``Request::setTrustedHosts()`` method::
+
+    // web/app.php
+    Request::setTrustedHosts(array('.*\.?acme.com$', '.*\.?acme.org$'));
+
+The default value for this option is an empty array, meaning that the application
+can respond to any given host.
+
+.. seealso::
+
+    Read more about this in the `Security Advisory Blog post`_.
+
+.. _reference-framework-form:
+
+form
+~~~~
+
+.. _reference-form-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the form services or not in the service container.
+
+This option will automatically be set to ``true`` when one of the child
+settings is configured.
+
+.. note::
+
+    This will automatically enable the validation.
+
+.. seealso::
+
+    For more details, see :doc:`/book/forms`.
+
+csrf_protection
+...............
+
+.. seealso::
+
+    For more information about CSRF protection in forms, see :ref:`forms-csrf`.
+
+.. _reference-form-csrf_protection-enabled:
+
+enabled
+"""""""
+
+**type**: ``boolean`` **default**: ``true``
+
+Whether to enable the CSRF support in forms or not. Setting this option to
+``true`` requires the `secret`_ to be set and `session`_ to be enabled. It
+becomes the service container parameter named ``form.type_extension.csrf.enabled``.
+
+This option will automatically be set to ``true`` when one of the child
+settings is configured.
+
+.. seealso::
+
+    Read more about CSRF in :ref:`the Book <forms-csrf>`.
+
+field_name
+""""""""""
+
+**type**: ``string`` **default**: ``'_token'``
+
+The name of the hidden field containing the CSRF token in the form. It becomes
+the service container parameter named ``form.type_extension.csrf.field_name``.
+
+esi
+~~~
+
+.. seealso::
+
+    You can read more about Edge Side Includes (ESI) in :ref:`edge-side-includes`.
+
+.. _reference-esi-enabled:
+
+enabled
+.......
+
+**type**: ``boolean`` **default**: ``false``
+
+Whether to enable the edge side includes support in the framework.
+
+You can also set ``esi`` to ``true`` to enable it:
```

Comment:
What this means: You can set `framework.esi.enabled` to `true`, or `framework.esi` to true.